src/abstractdataframe/reshape.jl

```diff
@@ -579,7 +580,13 @@ function _unstack(df::AbstractDataFrame, rowkeys::AbstractVector{Int},
     # and names between df1 and df2 must be unique
     # here df1 gets proper column-level metadata with :note style
     df1 = df[row_group_row_idxs, g_rowkey.cols]
-    df2 = DataFrame(unstacked_val, Symbol[Symbol(renamecols(x)) for x in colref_map],
+    new_col_names = Symbol[Symbol(renamecols(x)) for x in colref_map]
+    if !allunique(new_col_names)
+        throw(ArgumentError("Non-unique column names produced. " *
+                            "Non equal values in `colkey` were mapped " *
+                            "to the same column name."))
+    end
```

Comment:
In theory this is breaking. Can we be reasonably sure that there are no legitimate use cases?

Mixing `"a"` and `'a'` or `missing` and `"missing"` in the same column and expecting them to be treated the same seems relatively unlikely, but are there other possible cases?

Reply:
> In theory this is breaking

It is not breaking. This PR only adds more informative error messages. Currently, we have for example (other cases produce similar errors - which are quite confusing - this "confusing" nature of the error messages is the point of why this PR is made as we do not allow passing `makeunique=true`):

```
julia> df = DataFrame(x=1:3, y=["a", "b", "c"], z=11:13)
3×3 DataFrame
 Row │ x      y       z
     │ Int64  String  Int64
─────┼──────────────────────
   1 │     1  a          11
   2 │     2  b          12
   3 │     3  c          13

julia> unstack(df, :x, :y, :z, renamecols=x->"v")
ERROR: ArgumentError: Duplicate variable names: :v. Pass makeunique=true to make them unique using a suffix automatically.
```

> are there other possible cases?

The most likely cases are:
* when a user passes a column renaming function that produces duplicates (e.g. strips some part of the name and user thinks that it is still unique while it is not)
* when column name conflicts with key name.

Reply:
To expand on it. If `unstack` sees `'a'` and `"a"` it puts them in different groups (the behavior is just like `groupby` behaves) and performs unstacking. The error is produced much later when column names are generated for the final data frame that is ultimately returned.